src/sentry/monitors/endpoints/organization_monitors.py

```diff
@@ -69,6 +69,20 @@ def get(self, request: Request, organization: Organization) -> Response:
             )
         )
         query = request.GET.get("query")
+
+        if "environment" in filter_params:
+            monitor_environments = MonitorEnvironment.objects.filter(
+                environment=filter_params["environment_objects"][0]
+            )
+            prefetch = Prefetch(
+                "monitorenvironment_set",
+                queryset=monitor_environments,
+                to_attr="selected_monitorenvironment",
+            )
+            queryset = queryset.prefetch_related(prefetch).filter(
+                monitorenvironment__in=monitor_environments
+            )
```

Comment:
I'm not super confident in this approach, looking for some feedback

Reply:
Might be easier to do `.filter(monitorenvironment__environment=filter_params["environment_objects"][0])`